traceon/geometry.py

```diff
@@ -673,6 +670,126 @@ def aperture(height, radius, extent, z=0.):
         Path"""
         return Path.line([extent, 0., -height/2], [radius, 0., -height/2])\
                 .line_to([radius, 0., height/2]).line_to([extent, 0., height/2]).move(dz=z)
+
+    @staticmethod
+    def arc_polar(radius, angle, start=[0,0,1], plane_normal=[0,1,0], direction=[1,0,0]):
+        """Return an arc of specified by polar coordinates. The arc lies in a plane defined by the 
+        provided normal vector and curves from the start point in the specified direction 
+        counterclockwise around the normal.
+
+        Parameters
+        ---------------------------
+        radius : float
+            The radius of the arc.
+        angle : float
+            The angle subtended by the arc (in radians)
+        start: (3,) float
+            The start point of the arc
+        plane_normal : (3,) float
+            The normal vector of the plane containing the arc
+        direction : (3,) float
+            A tangent of the arc at the starting point. 
+            Must lie in the specified plane. Does not need to be normalized. 
+        Returns
+        ----------------------------
+        Path"""
+        start = np.array(start, dtype=float)
+        plane_normal = np.array(plane_normal, dtype=float)
+        direction = np.array(direction, dtype=float)
+
+        dot_product = np.dot(direction, plane_normal)
+
+        if not np.isclose(dot_product, 0, atol=1e-7):
+            corrected_direction = direction - (dot_product / np.linalg.norm(plane_normal)**2) * plane_normal
+            raise AssertionError(
+                f"The provided direction {direction} does not lie in the specified plane. \n"
+                f"The closed valid direction is {np.round(corrected_direction, 10)}.")
+        
+        direction /= np.linalg.norm(direction)
+        plane_normal /= np.linalg.norm(plane_normal)
```

Comment:
Consider doing these division before the dot product, this way your `atol=1e-7` is not affected when the vectors are very large or very small

Reply:
yes good point, I could do that. I currently set it up this way so that the AssertionError referred to the input vectors, which I think is more natural and interpretable for the user. I will just create a separate normalized version instead of dividing in place.